File: slot-dart/lib/src/models/reels_model.dart
library slot.models.reels_model;

import 'package:slot/src/mvc/model.dart';
import 'icon_model.dart';
import 'reel_model.dart';

class ReelsModel extends Model {

  List<List<int>> _reelsConfig;
  List<ReelModel> _reelModels;

  ReelsModel(List<List<int>> reelsConfig) : super() {
    _reelsConfig = reelsConfig;
    _reelModels = new List<ReelModel>();

    _createReelModels();
  }

  void _createReelModels() {
    for (List<int> reelConfig in _reelsConfig) {
      List<IconModel> iconModels = new List<IconModel>();

      for (int iconId in reelConfig) {
        iconModels.add(new IconModel(iconId));
      }

      ReelModel reelModel = new ReelModel(iconModels);

      _reelsModel.add(reelModel);
    }
  }

  List<ReelModel> get reelModels {
    return _reelModels;
  }

}

File: slot-ts/src/models/ReelsModel.ts
import IconModel from '../models/IconModel';
import ReelModel from '../models/ReelModel';
import Model from '../engine/mvc/Model';

export default class ReelsModel extends Model {
    private reelsConfig: number[][];
    private reelModels: ReelModel[];

    constructor(reelsConfig: number[][]) {
        super();
        this.reelsConfig = reelsConfig;
        this.reelModels = [];
        this.createReelModels();
    }

    private createReelModels(): void {
        for (const reelConfig of this.reelsConfig) {
            const iconModels: IconModel[] = [];
            for (const iconId of reelConfig) {
                iconModels.push(new IconModel(iconId));
            }
            const reelModel: ReelModel = new ReelModel(iconModels);
            this.reelModels.push(reelModel);
        }
    }

    public  getReelModels(): ReelModel[] {
        return this.reelModels;
    }
}

